File: src/www/__test__/testPost/helpers/PLSEmptyLine.ts
import { PostLineStep } from "../PostLineStep";
import { PostLine } from "../reader";

export class PLSEmptyLine implements PostLineStep {
  match(line: PostLine): boolean {
    return !!line.match(/^\s*$/);
  }

  async run(line: PostLine) {
    // Do nothing
  }

  getPrettyPrint(): string {
    return `Ingore lines that are all spaces (${this.constructor.name})`;
  }
}

File: src/test/java/com/drpicox/game/testPost/helpers/PLSEmptyLine.java
package com.drpicox.game.testPost.helpers;

import com.drpicox.game.testPost.PostLineStep;
import com.drpicox.game.testPost.reader.PostLine;
import org.springframework.stereotype.Component;

@Component
public class PLSEmptyLine implements PostLineStep {
    @Override
    public boolean match(PostLine line) {
        return line.match("^\\s*$") != null;
    }

    @Override
    public void run(PostLine line) {
        // Do nothing
    }

    @Override
    public String getPrettyPrint() {
        return "Ignore lines that are all spaces (" + getClass().getName() + ")";
    }
}
